### PR TITLE
fix(sidebar): get last message and display it in sidebar

### PR DESCRIPTION
**Pull request by AlexZakablukov**
**What this PR does** 📖
get last message and display it in sidebar

**Which issue(s) this PR fixes** 🔨
[AP-1410](https://satellite-im.atlassian.net/browse/AP-1410)

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

**Comment by phillsatellite** [comment]
Is there anything we can do about all the loading bars appearing when user logs in? Sara and I we're both thinking it looked kinda messy 
https://user-images.githubusercontent.com/93608357/174358463-a2a513dd-be46-4cb1-8589-a72e409cf97f.mov

**Comment by stavares843** [comment]
agree

**Comment by molimauro** [comment]
We can change the loader or we can remove it and show the last msg we have in store (even though can be that is not the last) and once the real last is fetched show finally that

**Comment by AlexZakablukov** [comment]
Fixed, now it shows the last message from the store while the actual last message is loading, without any loaders

**Comment by Clayton215** [comment]
The sidebar works as should, however, I am not able to send pictures out of local. They show as sent on the local account but they don't actually go through. 
Screen Shot 2022-06-21 at 7 07 16 PM

**Review by josephmcg — CHANGES_REQUESTED** [review]
whats the goal of switching it from the textile store namespace to friends? I thought we already had a decent foundation and would just build that out

**Comment by AlexZakablukov** [comment]
> whats the goal of switching it from the textile store namespace to friends? I thought we already had a decent foundation and would just build that out

I made it because we use friends and groups but not conversations in sidebar, when we add a new friend conversation does not create and till we enter the chat a conversation will not create, but a friend can write us and the last message should be 
 changed. 
But if you think it is the wrong way, I can return it to textile

**Comment by josephmcg** [comment]
Thanks Alex, I'm honestly not sure what the 'correct place' for it is. I think it would be ideal if it could be more centralized, rather than spread across multiple store modules. but that's probably a separate discussion altogether

**Comment by molimauro** [comment]
Sara has the same problem too, @AlexZakablukov can you double check please?

**Comment by stavares843** [comment]
correct tried here, and was the same

I was able to send images but not receiving them 🔨

**Comment by AlexZakablukov** [comment]
done, check pls)

File: store/textile/getters.ts [hunk 1]
```diff
@@ -12,6 +12,14 @@ const getters = {
       ? Object.values(state.conversations[address].messages)
       : []
   },
+  getConversationLastMessage:
+    (state: TextileState, getters: any) => (address: string) => {
+      const conversation = getters.getConversation(address)
+      if (!conversation) {
+        return null
+      }
```
Comment (molimauro) [inline store/textile/getters.ts:20]: this can be removed since we have the `?`

File: components/views/user/User.html [hunk 2]
```diff
@@ -9,6 +9,7 @@
   <div class="user-info">
     <TypographyTitle :size="6" data-cy="sidebar-user-name" :text="user.name" />
     <div
+      v-if="lastMessage"
```
Comment (molimauro) [inline components/views/user/User.html:12]: isnt this always true?

File: store/textile/mutations.ts [hunk 3]
```diff
@@ -168,6 +168,17 @@ const mutations = {
...
+  setConversationLastMessage(
+    state: TextileState,
+    {
+      conversationId,
+      lastMessage,
+    }: { conversationId: string; lastMessage: MessageTrackerValues | null },
+  ) {
+    const conversations = { ...state.conversations }
+    conversations[conversationId].lastMessage = lastMessage
+    state.conversations = conversations
```
Comment (molimauro) [inline store/textile/mutations.ts:180]: are you spreading for reactivity right?

Comment (AlexZakablukov) [reply store/textile/mutations.ts:180]: Yes, I don't want to mutate the state

File: components/views/user/User.html (hunk 2 above)
Comment (AlexZakablukov) [reply components/views/user/User.html:12]: you are right

File: store/textile/mutations.ts (hunk 3 above)
Comment (molimauro) [reply store/textile/mutations.ts:180]: Ok perfect then!

File: store/textile/getters.ts (hunk 1 above)
Comment (AlexZakablukov) [reply store/textile/getters.ts]: fixed

File: components/views/user/User.html (hunk 2 above)
Comment (molimauro) [reply components/views/user/User.html:12]: still can see this

Comment (AlexZakablukov) [reply components/views/user/User.html]: changed

**Comment by stavares843** [comment]
https://user-images.githubusercontent.com/29093946/176716230-a73f044c-489d-45ef-bafe-856eac7da15b.mov

the messages were both showing 1:00 as a timestamp until user opened the conversation

**Comment by AlexZakablukov** [comment]
fixed

**Review by phillsatellite — APPROVED** [review]

**Review by molimauro — APPROVED** [review]

**Comment by WanderingHogan** [comment]
Closing this one and leaving the branch for now. We can come back to it when we get to it. This PR touches on a major change that iridium brings - the textile store and messages being stored in indexeddb are going away. We can come back to this close PR and branch though when we are able to port any needed changes to iridium

**Pull request closed without merging**